spring-integration-core/src/main/java/org/springframework/integration/endpoint/AbstractPollingEndpoint.java

```diff
@@ -343,13 +345,16 @@ public void run() {
 						count++;
 					}
 					catch (Exception e) {
-						if (e instanceof RuntimeException) {
-							throw (RuntimeException) e;
+						if (e instanceof MessagingException) {
+							throw (MessagingException) e;
 						}
 						else {
-							throw new MessageHandlingException(new ErrorMessage(e), e);
+							throw new MessagingException(messageHolder.get(), e);
```

Comment:
Well, that is fully different story (enhanced exception) and let's consider it in the separate task!

And well, let it be just general `ME`!